botocore/validate.py

```diff
@@ -181,6 +188,25 @@ def _validate_jsonvalue_string(self, params, shape, errors, name):
         except (ValueError, TypeError) as e:
             errors.report(name, 'unable to encode to json', type_error=e)
 
+    def _validate_document(self, params, shape, errors, name):
```

Comment:
+1, this method is critical for making sure we can evolve document types as needed. For example, we will likely eventually add support for serializing and (lazily) deserializing modeled values in and out of document types. I don't want us to be blocked on that feature due to any implicit behavior in how Python serializes values (like if it checks to_json methods or something implicit).